app/components/selective_button/selective_button.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import {TouchableOpacity, Text, View} from 'react-native';
+import {VIEW_STYLES, TEXT_STYLES} from './selective_button.presets';
+import {SelectiveButtonProps} from './selective_button.props';
+
+/**
+ * For your text displaying needs.
+ *
+ * This component is a HOC over the built-in React Native one.
+ */
+
+export function SelectiveButton(props: SelectiveButtonProps) {
+  // grab the props
+  const {
+    onPreset = 'primary',
+    offPreset = 'buttonOff',
+    text,
+    style: styleOverride,
+    textStyle: textStyleOverride,
+    children,
+    ...rest
+  } = props;
+
+  const [isOn, setOn] = React.useState(false);
+  function toggle() {
+    setOn(!isOn);
+  }
+  const viewStyle =
+    VIEW_STYLES[isOn ? props.onPreset : props.offPreset] || VIEW_STYLES.primary;
```

Comment:
it should be onPreset because you already have it above
```suggestion
    VIEW_STYLES[isOn ? onPreset : offPreset] || VIEW_STYLES.primary;
```